pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/ColumnMetadataImpl.java

```diff
@@ -335,6 +378,22 @@ public static FieldSpec generateFieldSpec(String column, PropertiesConfiguration
     return fieldSpec;
   }
 
+  // This method is only meant to retain compatibility of deserialization for
+  // /tables/{tableName}/segments/{segmentName}/metadata endpoint .
+  @SuppressWarnings("unused")
+  public Map<IndexType<?, ?, ?>, Long> getIndexSizeMap() {
+    IndexService service = IndexService.getInstance();
+    Map<IndexType<?, ?, ?>, Long> result = new HashMap<>();
+
+    for (int i = 0, n = getNumIndexes(); i < n; i++) {
```

Comment:
(minor) Same here, you can directly loop over `_indexTypeSizeList` to reduce access